src/MicroOrm.Dapper.Repositories/ReadOnlyDapperRepository.cs

```diff
@@ -36,10 +39,18 @@ public ReadOnlyDapperRepository(IDbConnection connection, ISqlGenerator<TEntity>
         }
 
         /// <inheritdoc />
-        public IDbConnection Connection { get; set; }
+        public IDbConnection Connection
+        {
+            get => _connection ?? throw new ArgumentNullException(nameof(_connection));
```

Comment:
I think we don't need this. Connection passed in the constructor and can't be bull already

Reply:
```c#
public IDbConnection Connection { get; set; }
```

Reply:
`Connection` will be set null on `Dispose()`, throw `ObjectDisposedException` now.